stablehlo/tests/ops_speculatability.mlir

```diff
@@ -0,0 +1,24 @@
+// RUN: stablehlo-opt %s --hlo-test-speculatability --split-input-file --allow-unregistered-dialect | FileCheck %s
+
+// CHECK-LABEL: func @dynamic_iota
+// CHECK-NEXT:  return
+func.func @dynamic_iota(%unknown_shape: tensor<2xi32>) {
+  %constant_shape = stablehlo.constant dense<[3, 4]> : tensor<2xi32>
+  %speculatable_0 = stablehlo.dynamic_iota %constant_shape, dim = 0 : (tensor<2xi32>) -> tensor<3x4xi64>
+  %speculatable_1 = stablehlo.dynamic_iota %constant_shape, dim = 0 : (tensor<2xi32>) -> tensor<3x?xi64>
+  %speculatable_2 = stablehlo.dynamic_iota %constant_shape, dim = 0 : (tensor<2xi32>) -> tensor<?x4xi64>
+  %speculatable_3 = stablehlo.dynamic_iota %constant_shape, dim = 0 : (tensor<2xi32>) -> tensor<?x?xi64>
+  %not_speculatable_0 = stablehlo.dynamic_iota %unknown_shape, dim = 0 : (tensor<2xi32>) -> tensor<3x4xi64>
+  %not_speculatable_1 = stablehlo.dynamic_iota %unknown_shape, dim = 0 : (tensor<2xi32>) -> tensor<3x?xi64>
+  %not_speculatable_2 = stablehlo.dynamic_iota %unknown_shape, dim = 0 : (tensor<2xi32>) -> tensor<?x4xi64>
+  %speculatable_4 = stablehlo.dynamic_iota %unknown_shape, dim = 0 : (tensor<2xi32>) -> tensor<?x?xi64>
```

Comment:
can we put is next to `speculatable_3`?

Reply:
IMO it's more readable this way because this makes it easier to verify that all the cases are covered. In order:
1. shape argument: constant, unknown
2. dimensions: 3x4, 3x?, ?x4, ?x?

WDYT?